pybluetooth/pyusb_bt_sockets.py

```diff
@@ -0,0 +1,194 @@
+        if (d.bDeviceClass == USB_DEVICE_CLASS_WIRELESS_CONTROLLER and
+            d.bDeviceSubClass == USB_DEVICE_SUB_CLASS_RF_CONTROLLER and
+            d.bDeviceProtocol == USB_DEVICE_PROTOCOL_BLUETOOTH):
+            return True
+
+        # Check if it's a composite device:
+        if not (d.bDeviceClass == USB_DEVICE_CLASS_MISCELLANEOUS and
+                d.bDeviceSubClass == USB_DEVICE_SUB_CLASS_COMMON_CLASS and
+                d.bDeviceProtocol == USB_DEVICE_PROTOCOL_IAD):
+            return False
+        for cfg in d:
+            bt_intf_descr = {
+                "bInterfaceClass": USB_DEVICE_CLASS_WIRELESS_CONTROLLER,
+                "bInterfaceSubClass": USB_DEVICE_SUB_CLASS_RF_CONTROLLER,
+                "bInterfaceProtocol": USB_DEVICE_PROTOCOL_BLUETOOTH,
+            }
+            intf = usb.util.find_descriptor(cfg, **bt_intf_descr)
+            if intf is not None:
+                return True
+        return False
+
+    devs = []
+    matchers = [CUSTOM_USB_DEVICE_MATCHER, bt_adapter_matcher]
+    for matcher in matchers:
+        if not matcher:
+            continue
+        devs += list(usb.core.find(find_all=True, custom_match=matcher))
+    return devs
+
+
+class PyUSBBluetoothNoAdapterFoundException(Exception):
+    pass
+
+
+def find_first_bt_adapter_pyusb_device_or_raise():
+    pyusb_devs = find_all_bt_adapters()
+    if len(pyusb_devs) == 0:
+        raise PyUSBBluetoothNoAdapterFoundException(
+            "No Bluetooth adapters found!")
+
+    def _is_usable_device(pyusb_dev):
+        try:
+            pyusb_dev.set_configuration()
+            PyUSBBluetoothHCISocket(pyusb_dev).hci_reset()
+            return True
+        except:
+            return False
+
+    pyusb_devs = filter(_is_usable_device, pyusb_devs)
+
+    if len(pyusb_devs) == 0:
+        raise PyUSBBluetoothNoAdapterFoundException(
+            "No Bluetooth *usable* adapters found!")
+
+    if len(pyusb_devs) > 1:
+        LOG.warn("More than 1 Bluetooth adapters found, "
+                 "using the first one...")
+    pyusb_dev = pyusb_devs[0]
+
+    return pyusb_dev
+
+
+def find_first_bt_adapter_pyusb_device():
+    try:
+        return find_first_bt_adapter_pyusb_device_or_raise()
+    except PyUSBBluetoothNoAdapterFoundException:
+        return None
+
+
+def has_bt_adapter():
+    pyusb_dev = find_first_bt_adapter_pyusb_device()
+    if pyusb_dev is None:
+        return False
+    usb.util.dispose_resources(pyusb_dev)
+    return True
+
+
+def pebble_usb_class_matcher(d):
+    """ USB device class matcher for Pebble's Test Automation dongles """
+    USB_DEVICE_CLASS_VENDOR_SPECIFIC = 0xFF
+    USB_DEVICE_SUB_CLASS_PEBBLE_BT = 0xBB
+    USB_DEVICE_PROTOCOL_PEBBLE_BT = 0xBB
+    return (d.bDeviceClass == USB_DEVICE_CLASS_VENDOR_SPECIFIC and
+            d.bDeviceSubClass == USB_DEVICE_SUB_CLASS_PEBBLE_BT and
+            d.bDeviceProtocol == USB_DEVICE_PROTOCOL_PEBBLE_BT)
+
+CUSTOM_USB_DEVICE_MATCHER = pebble_usb_class_matcher
```

Comment:
Added this, even though it's Pebble-specific I decided it makes development of this lib much less painful for ourselves if we've got our matcher here.